File: data/readable_format/fortran/file656.f90
! square.f90
module math_operations
  implicit none
contains
  function square(x) result(res)
    implicit none
    real, intent(in) :: x
    real :: res
    res = x * x
  end function square
end module math_operations

program test
  use math_operations
  implicit none
  real :: number, result
  number = 4.0
  result = square(number)
  print *, "Square of", number, "is", result
end program test

File: data/readable_format/cpp/file656.cpp
// square.cpp
#include <iostream>
#include <cassert> // For assert

// Function prototype
float square(float x);

int main() {
    float number = 4.0f;
    float result = square(number);
    std::cout << "Square of " << number << " is " << result << std::endl;
    
    // Simple unit test
    assert(square(2.0f) == 4.0f); // This should pass
    assert(square(-3.0f) == 9.0f); // This should also pass
    
    std::cout << "All tests passed." << std::endl;
    return 0;
}

// Function definition
float square(float x) {
    return x * x;
}